packages/rspack/src/rspackOptionsApply.ts

```diff
@@ -135,6 +135,10 @@ export function optionsApply_compat(
 			compiler.context,
 			options.entry
 		);
+
+		if (options.devServer?.hot) {
```

Comment:
need to move to dev-server, move to here is for avoiding break change, maybe we need a future flag, or just use webpack-dev-server and deprecate rspack-dev-server